src/Microsoft.Kusto.ServiceLayer/DataSource/Contracts/KustoConnectionDetails.cs

```diff
@@ -0,0 +1,11 @@
+namespace Microsoft.Kusto.ServiceLayer.DataSource.Contracts
+{
+    public class KustoConnectionDetails
```

Comment:
New model for passing connection details specific to the DataSource. Atm, mapping is done in the DataSourceFactory